src/main/java/org/folio/circulation/domain/RequestQueueRepository.java

```diff
@@ -94,4 +98,17 @@ public static RequestQueueRepository using(Clients clients) {
     }
     return requestUpdated.thenApply(r -> r.map(notUsed -> requestQueue));
   }
+
+  CompletableFuture<Result<RequestQueue>> reorderRequests(RequestQueue requestQueue) {
+    Collection<Request> requestsWithChangedPosition = requestQueue
+      .getRequestsWithChangedPosition();
+
+    if (requestsWithChangedPosition.isEmpty()) {
+      LOG.info("No requests with changed positions found");
+      return completedFuture(succeeded(requestQueue));
+    }
+
+    return requestRepository.batchUpdate(requestsWithChangedPosition)
+      .thenApply(r -> r.map(result -> requestQueue));
+  }
```

Comment:
I was hoping all request queue reorder methods would be coalesced as part of this PR, but, now that I think about it, it is probably an unreasonable expectation. Anyway, I don't see why we can't. We should create a tech debt story for this.

Reply:
Since the existing method is used in various cases (fulfillment, checkout, etc.) I think it make sense to work on this in a separate story, so we can better analyze/catch any regression issues, just in case.

Reply:
I've created https://issues.folio.org/browse/CIRC-527

Reply:
@bohdan-suprun Thanks for raising the follow up issue